### PR TITLE
Create RepositoryBadge

### PR DESCRIPTION
**Pull request by benediktvaldez**
Not sure if these fields are the best option - just an idea

github repository_badge

### CODE REVIEW

File: src/components/RepositoryBadge.jsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            repository: null
+        };
+    }
+
+    getApiRequest() {
+        let { repository } = this.props;
+
+        return {
+            id:     `github.repository.${ repository }`,
+            params: {
```
Comment (plouc) [inline src/components/RepositoryBadge.jsx]:
``` javascript
params: { repository }
```

File: src/components/RepositoryBadge.jsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                repository: repository
+            }
+        };
+    }
+
+    onApiData(repository) {
+        this.setState({
```
Comment (plouc) [inline src/components/RepositoryBadge.jsx]:
``` javascript
this.setState({ repository });
```

File: src/components/RepositoryBadge.jsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+class RepositoryBadge extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            repository: null
+        };
+    }
+
+    getApiRequest() {
+        let { repository } = this.props;
```
Comment (plouc) [inline src/components/RepositoryBadge.jsx]: `const`

File: src/components/RepositoryBadge.jsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+    onApiData(repository) {
+        this.setState({
+            repository: repository
+        });
+    }
+
+    render() {
+        let { repository, title } = this.props;
+        var titleNode = (
```
Comment (plouc) [inline src/components/RepositoryBadge.jsx]: `const`

File: src/components/RepositoryBadge.jsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            <span>
+                <span className="widget__header__subject">{title || repository}</span>
+            </span>
+        );
+        var repositoryNode = (
```
Comment (plouc) [inline src/components/RepositoryBadge.jsx]: `let`

File: src/components/RepositoryBadge.jsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+RepositoryBadge.propTypes = {
+    repository: PropTypes.string.isRequired
+};
+
+reactMixin(RepositoryBadge.prototype, ListenerMixin);
+reactMixin(RepositoryBadge.prototype, Mozaik.Mixin.ApiConsumer);
+
+export { RepositoryBadge as default };
```
Comment (plouc) [inline src/components/RepositoryBadge.jsx:92]:
``` javascript
export default RepositoryBadge;
```

File: src/tests/RepositoryBadge.test.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
+import React        from 'react/addons';
+const { TestUtils } = React.addons;
+import { expect }   from 'chai';
+import sinon        from 'sinon';
+import mockery      from 'mockery';
+
+var RepositoryBadge;
```
Comment (plouc) [inline src/tests/RepositoryBadge.test.js:7]: `let`

File: src/tests/RepositoryBadge.test.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+var repositoryBadge;
```
Comment (plouc) [inline src/tests/RepositoryBadge.test.js:8]: `let`

File: src/tests/RepositoryBadge.test.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        expect(repositoryBadge.getApiRequest()).to.eql({
+            id:     'github.repository.plouc/mozaik',
+            params: {
+                repository: 'plouc/mozaik'
+            }
+        });
+    });
+
+    it('should display repository name', () => {
+        let repositoryname = TestUtils.findRenderedDOMComponentWithClass(repositoryBadge, 'widget__header');
```
Comment (plouc) [inline src/tests/RepositoryBadge.test.js:48]: `const`

File: src/components/RepositoryBadge.jsx [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+            </span>
+        );
+        var repositoryNode = (
+            <div className="widget__body" />
+        );
+
+        if (this.state.repository) {
+            titleNode = title === undefined ? (
+                <span>
+                    <span className="widget__header__subject">{this.state.repository.name}</span>
```
Comment (plouc) [inline src/components/RepositoryBadge.jsx]: I think it should also include repository owner.

File: src/components/RepositoryBadge.jsx [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+                <span className="widget__header__subject">{title || repository}</span>
+            </span>
+        );
+        var repositoryNode = (
+            <div className="widget__body" />
+        );
+
+        if (this.state.repository) {
+            titleNode = title === undefined ? (
+                <span>
```
Comment (plouc) [inline src/components/RepositoryBadge.jsx]: why do you add an extra `<span/>`?

File: src/components/RepositoryBadge.jsx [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    onApiData(repository) {
+        this.setState({
+            repository: repository
+        });
+    }
+
+    render() {
+        let { repository, title } = this.props;
```
Comment (plouc) [inline src/components/RepositoryBadge.jsx:30]: `const`

File: src/components/RepositoryBadge.jsx [hunk 13]
```diff
@@ -0,0 +1,105 @@
+import React, { Component, PropTypes } from 'react';
+import reactMixin                      from 'react-mixin';
+import { ListenerMixin }               from 'reflux';
+import _                               from 'lodash';
```
Comment (plouc) [inline src/components/RepositoryBadge.jsx:4]: not used

File: src/components/RepositoryBadge.jsx [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+            <div className="widget__body" />
+        );
+
+        if (this.state.repository) {
+            titleNode = title === undefined ? (
+                <span>
+                    <span className="widget__header__subject">{this.state.repository.name}</span>
+                </span>
+            ) : title;
+            var date = this.state.repository.updated_at.split('T')[0];
```
Comment (plouc) [inline src/components/RepositoryBadge.jsx:42]: `const`

Comment (plouc) [inline src/components/RepositoryBadge.jsx:42]: perhaps [this](http://momentjs.com/docs/#/displaying/fromnow/) is a good option

File: src/components/RepositoryBadge.jsx [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+                        <span>{this.state.repository.description}</span>
+                    </div>
+                    <div className="github__repository-badge__info">
+                        <div className="github__repository-badge__info__item">
+                            <span className="count">{this.state.repository.open_issues_count}</span>&nbsp;
+                            issues
+                        </div>
+                        <div className="github__repository-badge__info__item">
+                            <span className="count">{this.state.repository.size}</span>&nbsp;
+                            size
```
Comment (plouc) [inline src/components/RepositoryBadge.jsx:55]: A unit could be helpful

File: src/tests/RepositoryBadge.test.js [hunk 16]
```diff
@@ -0,0 +1,51 @@
+import React        from 'react/addons';
+const { TestUtils } = React.addons;
+import { expect }   from 'chai';
+import sinon        from 'sinon';
+import mockery      from 'mockery';
+
+var RepositoryBadge;
+var repositoryBadge;
+
+describe('Github — RepositoryBadge', () => {
```
Comment (plouc) [inline src/tests/RepositoryBadge.test.js:10]: This test is perhaps a little bit incomplete ;)

**Comment by plouc** [comment]
@benediktvaldez, it could be nice to use [icons](https://fortawesome.github.io/Font-Awesome/icons/) for each fields, **font-awseome** is already included, you just have to create an element like:

``` html
<i className="fa fa-check" />
```

**Comment by plouc** [comment]
We could add an icon in the header to, using github icon and fork icon if the repo is a forked one

File: src/components/RepositoryBadge.jsx (hunk 10 above)
Comment (plouc) [inline src/components/RepositoryBadge.jsx]: and keep `repository.name` only with the `widget__header__subject` class

**Comment by plouc** [comment]
I'll add fork count to

File: src/components/RepositoryBadge.jsx (hunk 11 above)
Comment (benediktvaldez) [inline src/components/RepositoryBadge.jsx]: Copied this from some existing widget - not sure why :/

**Comment by benediktvaldez** [comment]
Wow, great feedback @plouc - I'll work on these and update this branch

**Pull request closed without merging**